### ISSUE TITLE
Can't pull images when running in Containers for Windows

### ISSUE THREAD

**Issue by francoisdutoit** [dotnet/dotnet-docker#992]
## Steps to reproduce the issue
1. On Windows 10 - Switch to Containers for Windows

## Expected behavior
Pull .net Core images from DockerHub : docker pull mcr.microsoft.com/dotnet/core/sdk

## Actual behavior
Error:
read tcp 192.168.88.229:2897->204.79.197.219:443: wsarecv: An existing connection was forcibly closed by the remote host.

## Additional information (e.g. issue happens only occasionally)
When switching to Linux containers - images pull down without an issue

## Output of `docker version`
Client: Docker Engine - Community
 Version:           18.09.2
 API version:       1.39
 Go version:        go1.10.8
 Git commit:        6247962
 Built:             Sun Feb 10 04:12:31 2019
 OS/Arch:           windows/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.2
  API version:      1.39 (minimum version 1.24)
  Go version:       go1.10.6
  Git commit:       6247962
  Built:            Sun Feb 10 04:28:48 2019
  OS/Arch:          windows/amd64
  Experimental:     false

## Output of `docker info`

Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 0
Server Version: 18.09.2
Storage Driver: windowsfilter
 Windows:
Logging Driver: json-file
Plugins:
 Volume: local
 Network: ics l2bridge l2tunnel nat null overlay transparent
 Log: awslogs etwlogs fluentd gelf json-file local logentries splunk syslog
Swarm: inactive
Default Isolation: hyperv
Kernel Version: 10.0 17134 (17134.1.amd64fre.rs4_release.180410-1804)
Operating System: Windows 10 Pro Version 1803 (OS Build 17134.648)
OSType: windows
Architecture: x86_64
CPUs: 8
Total Memory: 15.96GiB
Name: FrancoisDesktop
ID: 3BZN:MCQK:E4YF:JHFQ:45A5:5NQS:7EGG:Q7AF:R4AK:RWF5:GR5U:SOEN
Docker Root Dir: C:\ProgramData\Docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: -1
 Goroutines: 26
 System Time: 2019-03-17T06:32:46.5001332+02:00
 EventsListeners: 1
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
Product License: Community Engine

**Comment by MichaelSimons** [comment]
@francoisdutoit - Can you pull any images from mcr.microsoft.com (e.g. `docker pull mcr.microsoft.com/windows/nanoserver:1809`)?

@rotata - Do you know anything about this?

**Comment by francoisdutoit** [comment]
@MichaelSimons - I am able to pull the Microsoft images for Linux - so when Docker is running in Containers for Linux - all is well .When I however switch over to Containers for Windows I am unable to pull the images.

![image](https://user-images.githubusercontent.com/9386768/54543292-10fe2280-49a6-11e9-926c-3d3d3dda5ee7.png)

**Comment by asmichi** [comment]
https://twitter.com/asm_ichi/status/1124535074429366273
> read tcp 192.168.1.65:27527->204.79.197.219:443: wsarecv: An existing connection was forcibly closed by the remote host.

I experienced the same problem. It seems to me the same issue as moby/moby#33667.

TL;DR: Yesterday pull from 204.79.197.219 (winlayers.cdn.mscr.io) failed from a Japan East Azure VM. No problem from a West US VM.

Yesterday I tried to `docker pull mcr.microsoft.com/dotnet/framework/runtime:4.7.2-windowsservercore-ltsc2019` more than 10 times but all attempts resulted in the same error.

I created an Azure VM in the *Japan East* region and tried the pull command. The command failed with a slightly different error (I forgot the details). Then I tried the pull command from a VM in the *West US* region and it pulled the image with no error. (Probably connection speed matters??)

Around 2019-05-04 02:00 PST, suddenly the server came back good and I succeeded in `docker pull mcr.microsoft.com/dotnet/framework/runtime:4.7.2-windowsservercore-ltsc2019` with no error.

Seems to me that sometimes the CDN server (204.79.197.219, winlayers.cdn.mscr.io) goes wrong for a while, then comes back working.

Hope this helps Microsoft folks to identify the root problem.

A test script derived from moby/moby#33667: [Test-cdn.mscr.ps1.txt](https://github.com/dotnet/dotnet-docker/files/3145151/Test-cdn.mscr.ps1.txt)
